### PR TITLE
update api implementation and generate client

### PR DESCRIPTION
**Pull request by Rjected**
## Major changes
  - generated client
  - updated dockerfile version
  - create tests and benchmarks

### CODE REVIEW

File: galactuslib/api_room_service_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	// create first client
+	apic := roomapi.NewAPIClient(apiConfig)
+	background := context.Background()
+	retRoom, _, err := apic.RoomApi.AddRoom(background)
+	if err != nil {
+		t.Errorf("error adding room: %s", err)
+		return
+	}
+	t.Logf("room id: %d, room id str: %s\n", retRoom.Id, retRoom.Code)
+}
```
Comment (Rjected) [inline galactuslib/api_room_service_test.go:39]: Example of functioning service use in tests

File: cmd/roomservice/main.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/Galactus-Player/roomservice/galactuslib"
+)
+
+func main() {
+	log.Printf("Server started")
```
Comment (lukeyeh) [inline cmd/roomservice/main.go]: Maybe put this log after the server has started (line 27) or change the message to "Server starting" instead?

File: README.md [hunk 3]
```diff
@@ -1,25 +1,18 @@
...
 
 ### Running the server
 To run the server, follow these simple steps:
 
 ```
-go run main.go
+go run cmd/roomservice/main.go
 ```
 
 To run the server in a docker container
```
Comment (lukeyeh) [inline README.md]: Tests already show how to make calls to server, but maybe add a small snippet on how to do that here.

**Review by lukeyeh — CHANGES_REQUESTED** [review]
some very small nits, but LGTM

File: README.md (hunk 3 above)
Comment (lukeyeh) [reply README.md]: Or if you're lazy just add a link to the tests.

**Comment by lukeyeh** [comment]
LGTM

**Pull request merged**